src/components/evaluate.ts

```diff
@@ -0,0 +1,33 @@
+const expr = require('expression-eval');
+const cache = new Map();
+
+// Compile expression or return cached compiled expression
+function compile(str: string) {
+  const cached = cache.get(str);
+  if (cached) {
+    return cached;
+  } else {
+    const compiled = expr.compile(str);
+    cache.set(str, compiled);
+    return compiled;
+  }
+}
+
+export function evaluate(str: string, context: any, currentProp: string) {
+  // Transform `paths.*.*.responses.*.foo` into `foo`
+  const _context = {};
+  for (const prop in context) {
+    // Only consider properties ABOVE the current property in the schema
+    // (This enforces a top-to-bottom data dependency which is just nice.)
+    if (prop === currentProp) break;
+    const short = prop.split('.').pop() as string;
+    _context[short] = context[prop];
+  }
+  // Evaluate expression
+  try {
+    return compile(str)(_context);
+  } catch (err) {
+    console.log(err, str, _context);
+    return true;
```

Comment:
Why true and not false? :P

Reply:
I don't know whether we should track it or not... what if a user types an invalid expression?
We might receive tons of irrelevant reports.

Anyway, I'd (optionally) leave a comment that `true` is needed to make sure the field doesn't disappear or so, nothing more than that.

Reply:
(Remember users don't type the expressions. It would have to be some form value that causes the expression to fail.)

But regardless I made it configurable so it makes much more sense now.